File: Haskell/Problem_003.hs
-- Find the largest prime factor of a composite number.
import           Euler                          ( primes )


factors :: Integer -> [Integer]
factors 0 = []
factors 1 = []
factors n = factor : factors (n `div` factor)
  where factor = head $ filter (\p -> n `mod` p == 0) primes


main :: IO ()
main = do
  print $ last $ factors $ 600851475143

File: Python 3/Problem_003.py
# Find the largest prime factor of a composite number.
from math import sqrt, floor
from euler import primes


def max_factor(N):
    max_factor = None
    limit = floor(sqrt(N))

    for p in primes():
        if N % p == 0:
            max_factor = p
        if p > limit:
            return max_factor


print(max_factor(600851475143))
